routing_model/src/flows_dst.rs

```diff
@@ -387,30 +385,21 @@ impl ProcessElderChange {
     }
 
     fn routine_state(&self) -> &ProcessElderChangeState {
-        match &self
+        &self
             .0
             .check_and_process_elder_change_routine
             .sub_routine_process_elder_change
-        {
-            Some(state) => state,
-            _ => panic!("Expect ProcessElderChange {:?}", &self),
-        }
     }
 
     fn mut_routine_state(&mut self) -> &mut ProcessElderChangeState {
-        let clone = self.clone();
-        match &mut self
+        &mut self
             .0
             .check_and_process_elder_change_routine
             .sub_routine_process_elder_change
-        {
-            Some(state) => state,
-            _ => panic!("Expect ProcessElderChange {:?}", &clone),
-        }
     }
 
     fn mark_elder_change(&self) -> Self {
-        let change_elder = self.routine_state().change_elder.clone();
+        let change_elder = self.routine_state().change_elder.clone().unwrap();
```

Comment:
take?

Reply:
You mean changing it to?
```
fn mark_elder_change(&self) -> Self {
    let mut state = self.clone();
    let change_elder = state.mut_routine_state().change_elder.take().unwrap();
    state.0.action.mark_elder_change(change_elder);
    state
}      
```

Reply:
Yes. If you think it make sense.
Your version also make sense to me.

Reply:
actually forget it. Yours seem simpler.

Reply:
I'll use your second version with `take` :)

Reply:
They are slightly different though? Even though it might not be visible from the rest of the code. Feels more correct to leave change_elder as empty

Reply:
since you clean it up on exit, it seems ok to leave it. But you decide.

Reply:
Let's give the second reviewer the casting vote ;)

Reply:
I vote for `take()` :)